hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/ClusterJoinManager.java

```diff
@@ -120,27 +126,25 @@
         maxWaitMillisBeforeJoin = node.getProperties().getMillis(ClusterProperty.MAX_WAIT_SECONDS_BEFORE_JOIN);
         waitMillisBeforeJoin = node.getProperties().getMillis(ClusterProperty.WAIT_SECONDS_BEFORE_JOIN);
         staleJoinPreventionDuration = TimeUnit.SECONDS.toMillis(STALE_JOIN_PREVENTION_DURATION_SECONDS);
+        syncJoinStrategy = node.getProperties().getBoolean(ClusterProperty.WAIT_SECONDS_BEFORE_JOIN_ASYNC)
```

Comment:
I'd prefer if both strategies were in separate classes as implementations of an interface. Now, one strategy is in this class, which is a manager and another strategy is in another class, which uses this manager.

Reply:
I tried doing that, but it requires a lot more work. I am not sure if the older strategy is needed long-term so it's easier to remove